src/utils/risk.ts

```diff
@@ -0,0 +1,275 @@
+    const sortedData = data?.map(
+        (item) => {
+            if (isFalsyString(item.estimation_type) || item.estimation_type === '') {
+                return undefined;
+            }
+
+            if (isFalsyString(item.analysis_date) || isNotDefined(item.total_displacement)) {
+                return undefined;
+            }
+
+            return {
+                ...item,
+                estimation_type: item.estimation_type,
+                analysis_date: item.analysis_date,
+                estimation_priority: estimationPriorityMap[item.estimation_type],
+                analysisTimestamp: new Date(item.analysis_date).getTime(),
+                total_displacement: item.total_displacement,
+            };
+        },
+    ).filter(isDefined).sort((a, b) => (
+        compareNumber(a.year, b.year)
+        || compareNumber(a.month, b.month)
+        || compareNumber(a.estimation_priority, b.estimation_priority)
+        || compareNumber(a.analysisTimestamp, b.analysisTimestamp, -1)
+    )) ?? [];
+
+    const uniqueData = unique(
+        sortedData,
+        (ipcDataItem) => `${ipcDataItem.year}-${ipcDataItem.month}`,
+    );
+
+    return uniqueData;
+}
+
+export function getAverageIpcData(uniqueData: IpcData) {
+    const monthGroupedIpcData = listToGroupList(
+        uniqueData,
+        (datum) => datum.month,
+        (datum) => datum.total_displacement,
+    );
+
+    const ipcRiskDataItem = mapToMap(
+        monthGroupedIpcData,
+        (key) => monthNumberToNameMap[Number(key) - 1],
+        (item) => avgSafe(item),
+    );
+
+    const monthlyValueList = Object.values(ipcRiskDataItem ?? {});
+    const annual_average = avgSafe(monthlyValueList);
+
+    return {
+        ...ipcRiskDataItem,
+        annual_average,
+    };
+}
+
+export function getFiRiskDataItem(data: IpcData | undefined) {
+    if (isNotDefined(data) || data.length === 0) {
+        return undefined;
+    }
+
+    const uniqueData = getPrioritizedIpcData(data);
+    const ipcRiskDataItem = getAverageIpcData(uniqueData);
+    const firstData = data[0];
+
+    return {
+        hazard_type: 'FI' as const,
+        hazard_type_display: firstData?.hazard_type_display,
+        country_details: firstData?.country_details,
+        ...ipcRiskDataItem,
+    };
+}
+
+export function getWfRiskDataItem(data: GwisData | undefined) {
+    if (isNotDefined(data) || data.length === 0) {
+        return undefined;
+    }
+
+    const uniqueData = unique(
+        data,
+        (gwisDataItem) => `${gwisDataItem.year}-${gwisDataItem.month}`,
+    );
+
+    const monthGroupedGwisData = listToGroupList(
+        uniqueData,
+        (datum) => datum.month,
+        (datum) => datum.dsr_avg,
+    );
+
+    const gwisRiskDataItem = mapToMap(
+        monthGroupedGwisData,
+        (monthKey) => monthNumberToNameMap[Number(monthKey) - 1],
+        (item) => avgSafe(item),
+    );
+
+    const firstYearData = data[0];
```

Comment:
Why are we using the first data?

Reply:
To get the common elements of the items. We use any of the item from 0 to length of the list. 0 is just arbitrary here.

Reply:
Let's add NOTE